src/app/screens/LNURLRedeem/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { CaretLeftIcon } from "@bitcoin-design/bitcoin-icons-react/filled";
+import Button from "@components/Button";
+import Container from "@components/Container";
+import Header from "@components/Header";
+import IconButton from "@components/IconButton";
+import TextField from "@components/form/TextField";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useLocation, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import QrcodeAdornment from "~/app/components/QrcodeAdornment";
+import lnurlLib from "~/common/lib/lnurl";
+import { isLNURLDetailsError } from "~/common/utils/typeHelpers";
+
+function LNURLRedeem() {
+  const { t } = useTranslation("translation", { keyPrefix: "lnurlredeem" });
+  const location = useLocation();
+
+  const [lnurlwithdraw, setlnurlwithdraw] = useState(
+    location.state?.decodedText || ""
+  );
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(false);
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    try {
+      setLoading(true);
+      const lnurl = lnurlLib.findLnurl(lnurlwithdraw);
+
+      if (lnurl) {
+        const lnurlDetails = await lnurlLib.getDetails(lnurl);
+
+        if (isLNURLDetailsError(lnurlDetails)) {
+          toast.error(lnurlDetails.reason);
+          return;
+        }
+
+        if (lnurlDetails.tag === "withdrawRequest") {
+          navigate("/lnurlWithdraw", {
+            state: {
+              args: {
+                lnurlDetails,
+              },
+            },
+          });
+        } else {
+          toast.error(t("errors.invalid_withdraw_request"));
+          return;
+        }
+      } else {
+        toast.error(t("errors.invalid_lnurl"));
+        return;
+      }
+    } catch (e) {
+      if (e instanceof Error) {
+        toast.error(e.message);
+      }
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  return (
+    <div className="h-full flex flex-col overflow-y-auto no-scrollbar">
+      <Header
+        title={t("title")}
+        headerLeft={
+          <IconButton
+            onClick={() => navigate(-1)}
+            icon={<CaretLeftIcon className="w-4 h-4" />}
+          />
+        }
+      />
+      <form onSubmit={handleSubmit} className="h-full">
+        <Container justifyBetween maxWidth="sm">
+          <div className="pt-4">
+            <TextField
+              id="invoice"
```

Comment:
this is not an invoice, maybe the id should be renamed or removed if it is not necessary

Reply:
resolved